### PR TITLE
util-linux: update to 2.40.1

### PR DESCRIPTION
**Pull request by tanpsi**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64-GLIBC)

### CODE REVIEW

**Comment by tanpsi** [comment]
Fixed and tested for x86_64-musl

**Comment by tanpsi** [comment]
Fixed and tested for armv6l-musl

File: srcpkgs/util-linux/template [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
-revision=2
+version=2.40.1
+revision=1
 build_style=gnu-configure
 configure_args="--exec-prefix=\${prefix} --enable-libuuid --disable-makeinstall-chown
  --enable-libblkid --enable-fsck --disable-rpath --enable-fs-paths-extra=/usr/sbin:/usr/bin
  --enable-vipw --enable-newgrp --enable-chfn-chsh --with-systemdsystemunitdir=no
- --with-udev --without-python --enable-write --disable-libmount-mountfd-support"
+ --with-udev --without-python --enable-write --disable-libmount-mountfd-support
+ --disable-liblastlog2 --disable-pam-lastlog2"
```
Comment (classabbyamp) [inline srcpkgs/util-linux/template:11]: why disable lastlog2? I see no reason to

Comment (tanpsi) [reply srcpkgs/util-linux/template]: I read [this](https://karelzak.blogspot.com/) and thought that this is supposed to replace lastlog right away and would rather be done later and decided by the maintainers. Thanks for doing it!
Just for my information, will this not require changes to files in `/etc/pam.d/` for it to start logging in its db?

**Comment by tanpsi** [comment]
Just to mention, the currently packaged util-linux provides a `su` with a visible bug due to the `su-common-fix-ptr-truncate` patch. This was fixed [upstream](https://github.com/util-linux/util-linux/commit/4b2e6f5071a4c5beebbd9668d24dc05defc096d7).
This can be seen by a simple `su user` and `ps -f` on my system.

File: srcpkgs/util-linux/template (hunk 1 above)
Comment (classabbyamp) [reply srcpkgs/util-linux/template]: I think to use it it would require a change there, but I don't see any other distro adding it yet

**Comment by classabbyamp** [comment]
that would probably be worth backporting

**Pull request merged**